src/main.rs

```diff
@@ -187,10 +237,97 @@ fn byte_offset_to_point(file_contents: &Rope, byte_offset: usize) -> Point {
     }
 }
 
-fn tree_sitter_range_to_lsp_range(file_contents: &Rope, range: tree_sitter::Range) -> Range {
+fn byte_offset_to_position(file_contents: &Rope, byte_offset: usize) -> Position {
+    point_to_position(byte_offset_to_point(file_contents, byte_offset))
+}
+
+fn byte_offset_range_to_lsp_range(file_contents: &Rope, range: ops::Range<usize>) -> Range {
     Range {
-        start: point_to_position(byte_offset_to_point(file_contents, range.start_byte)),
-        end: point_to_position(byte_offset_to_point(file_contents, range.end_byte)),
+        start: byte_offset_to_position(file_contents, range.start),
+        end: byte_offset_to_position(file_contents, range.end),
+    }
+}
+
+fn tree_sitter_range_to_lsp_range(file_contents: &Rope, range: tree_sitter::Range) -> Range {
+    byte_offset_range_to_lsp_range(file_contents, range.start_byte..range.end_byte)
+}
+
+fn get_text_document_edits(
+    old_contents: &Rope,
+    new_contents: &Rope,
+    uri: &Url,
+) -> TextDocumentEdit {
+    let old_contents_as_str: Cow<'_, str> = old_contents.clone().into();
```

Comment:
Could probably implement text-diffing against `Rope`'s (maybe one approach there would be to modify `text_diff` to take "chunked" "inputs"?)

But for now doing the simplest thing which is to convert to `String`'s and then run an existing text-differ against those

Also as mentioned I think we could optimize the "only went through the fixing loop once" case and have it (where I guess "it" is inside `tree-sitter-lint`?) track the applied edit byte-ranges explicitly and return that/wire it up here (to avoid having to manually calculate diffs after-the-fact in what I'd guess is the "common case")